src/toil/job.py

```diff
@@ -59,11 +59,8 @@ def __init__(self, memory=sys.maxint, cores=sys.maxint, disk=sys.maxint):
         require to run. Cores is the number of CPU cores required.
         """
         self.cores = cores
-        # passing sys.maxint to human2bytes seems to result in some float imprecision, and returns a value 1
-        # larger than sys.maxint. We later assume that any value here not equal to sys.maxint must be the user's value
-        # so it is passed to the batch system, which cannot allocate that many resources.
-        self.memory = human2bytes(str(memory)) if memory!=sys.maxint else sys.maxint
-        self.disk = human2bytes(str(disk)) if disk!=sys.maxint else sys.maxint
+        self.memory = human2bytes(str(memory)) if memory else memory
```

Comment:
I'd prefer "is not None" otherwise the empty string would sneak through. With "is not None" you'd get an exception.